src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let phrase = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (typeof this[i] === 'number'
+      || typeof this[i] === 'string'
+      || typeof this[i] === 'boolean') {
```

Comment:
Will not work for bigInt, symbol, function, object
```suggestion
      if (this[i] !== undefined && this[i] !== null) {
```